plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java

```diff
@@ -1150,15 +1150,16 @@ public boolean downloadToFile(String urlString, Plugin plugin, @CheckForNull Fil
                 HttpGet httpGet = new HttpGet(urlString);
                 try {
                     httpclient.execute(httpGet, new FileDownloadResponseHandler(pluginFile), context);
-                    // get final URI (after all redirects)
-                    List<URI> locations = context.getRedirectLocations();
-                    if (locations != null) {
-                        logVerbose(String.format("Downloading %s from %s", plugin.getName(), locations.get(locations.size() - 1)));
-                    }
                 } catch (IOException e) {
                     logVerbose(String.format("Unable to resolve plugin URL %s, or download plugin %s to file",
                             urlString, plugin.getName()));
                     success = false;
+                } finally {
+                    // get final URI (after all redirects)
+                    List<URI> locations = context.getRedirectLocations();
+                    if (locations != null) {
+                        logVerbose(String.format("%s %s from %s", success ? "Downloaded" : "Tried downloading", plugin.getName(), locations.get(locations.size() - 1)));
```

Comment:
nice, if you're feeling extra amazing it would be great to log the failure case always and verbose for the regular case, but not required to ship this

Thanks!